src/ucp/core/ucp_context.c

```diff
@@ -2057,8 +2058,16 @@ ucp_context_dev_tl_bitmap(ucp_context_h context, const char *dev_name,
             continue;
         }
 
+        if (!(context->tl_rscs[tl_idx].flags & UCP_TL_RSC_FLAG_AUX)) {
+            has_non_aux_tl = 1;
+        }
```

Comment:
what if we replace "has_no_aux_tl" variable, by doing
```c
if (!(context->tl_rscs[tl_idx].flags & UCP_TL_RSC_FLAG_AUX)) {
   UCS_BITMAP_SET(*tl_bitmap, tl_idx);
}
```

so we set bit if have non-aux, instead of clearing the bit if did not find non-aux

Reply:
> if no DC-support and only auxiliary UD exist in case of UCX_TLS=dc, we should report an error to a user
Wouldn't there be an error from init_lanes?

Reply:
the idea is to do fallback in `ucp_cm_client_resolve_cb`

Reply:
won't we fallback if init-lanes failed?
IMO it's weird that bitmap zero indicates that all tls were aux.
should check if have non-aux tls outside this function

Reply:
> won't we fallback if init-lanes failed?

yes, it will fallback.
but we shouldn't print an error in this case - fixed by passing `show_error = 0` parameter in case of doing `init_lanes` for CM initial configuration.